crates/sui-network/src/discovery/mod.rs

```diff
@@ -145,28 +151,38 @@ impl DiscoveryEventLoop {
             peer_id: self.network.peer_id(),
             addresses: address,
             timestamp_ms: now_unix(),
+            access_type: self.discovery_config.access_type(),
         };
 
         self.state.write().unwrap().our_info = Some(our_info);
     }
 
     fn configure_preferred_peers(&mut self) {
-        for SeedPeer { peer_id, address } in self.config.seed_peers.iter() {
-            let Some(peer_id) = *peer_id else {
-                continue;
-            };
-
-            let Ok(address) = address.to_anemo_address() else {
-                debug!(p2p_address=?address, "Can't convert p2p address to anemo address");
-                continue;
+        for (peer_id, address) in self
+            .discovery_config
+            .allowlisted_peers
+            .iter()
+            .map(|sp| (sp.peer_id, sp.address.clone()))
+            .chain(self.config.seed_peers.iter().filter_map(|ap| {
+                ap.peer_id
+                    .map(|peer_id| (peer_id, Some(ap.address.clone())))
+            }))
+        {
+            let anemo_address = if let Some(address) = address {
+                let Ok(address) = address.to_anemo_address() else {
+                    debug!(p2p_address=?address, "Can't convert p2p address to anemo address");
+                    continue;
+                };
+                Some(address)
+            } else {
+                None
             };
 
             let peer_info = anemo::types::PeerInfo {
                 peer_id,
                 affinity: anemo::types::PeerAffinity::High,
```

Comment:
`PeerAffinity::High` "Always attempts to maintain a connection with this Peer" but I thought that's not what you wanted for the `allowlisted_peers`?

Reply:
Just to make sure we are on the same page and my understanding is correct:
* by marking a peer as `PeerAffinity::High` in `known_peers` does not **cause** a node to alway proactively reach out to the peer. But having that entry itself in `known_peers` does, as long as the peer is not `PeerAffinity::Never` ([code pointer](https://github.com/MystenLabs/anemo/blob/main/crates/anemo/src/network/connection_manager.rs#L328))
* Is the goal that we don't want a node to proactively reach out to a peer, but always accepts its connection?

Reply:
I thought the problem is simply we don't want a node to poke its seed peers in discovery. Can you help me understand what the real problem is?

Reply:
I thought the problem is we don't want, a node to always poke its seed peers in discovery, when the sole purpose of marking them as seed peers being allowlisting them in connection.

Reply:
According to the anemo documentation, `PeerAffinity::High` causes anemo to always maintain a connection with that peer. I think that's not what we want? We want to allow it inbound, but not proactively establish outbound connections with peers in the allowlisted set.

Reply:
You will probably need to add a new PeerAffinity::AllowUnlimited or something like that in anemo, I'm not sure what we need there exists rn

Reply:
I see. Yeah to achieve this we need a new `PeerAffinity` type, and in `handle_connectivity_check` we don't look to connect the allowlisted peers unless they are `PeerAffinity::High`. I will add that separately and change allowlisted peers in discovery to be `PeerAffinity::Allowlisted" or whatever it is when that's ready.

Reply:
That's fine, if you are wanting to merge this first and come back later to update could you add a TODO here indicating the plan to change this once anemo side is ready to not proactively connect to allowed peers?

Reply:
added!